lib/nss.c

```diff
@@ -18,16 +19,17 @@
 // the subids are a pretty limited resource, and local files seem
 // bound to step on any other allocations leading to insecure
 // conditions.
-static bool nss_initialized;
+static atomic_flag nss_init_started;
+bool nss_init_completed;
```

Comment:
Maybe `atomic_bool nss_init_completed`?

Reply:
I did that first, but then realized that there is no api that i could find for just checking the value or even clearing it.  Only set-and-check.  That makes this unsuitable for the loop where I'm just waiting for another thread to set it.

And it doesn't need to be atomix - only nss_init_started does.

Reply:
> I did that first, but then realized that there is no api that i could find for just checking the value or even clearing it.

Actually there are `atomic_load()`/`atomic_store()`.

But if I understand correctly, in case you don't need to specify explicit memory order, then `atomic_bool` can be just assigned/read (keeping "atomic" properties).

> And it doesn't need to be atomix - only nss_init_started does.

I think both have to. Otherwise compiler is eligible for different optimizations (for example, to cache value of `nss_init_completed` on a register while code is in the loop). In the past some folks used `volatile` to prevent it but this is... wrong.

https://en.cppreference.com/w/c/language/atomic

Reply:
Oh, I see - i had looked through stdatomix.h by the bool and flag definitions, but it's done through #defines.  Thanks